metacat-main/src/main/java/com/netflix/metacat/main/services/MViewServiceEventHandler.java

```diff
@@ -0,0 +1,108 @@
+
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.main.services;
+
+import com.netflix.metacat.common.NameDateDto;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.server.events.MetacatDeleteTablePostEvent;
+import com.netflix.metacat.common.server.events.MetacatRenameTablePostEvent;
+import com.netflix.metacat.common.server.properties.Config;
+import com.netflix.metacat.common.server.usermetadata.UserMetadataService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Event handler for view changes based on table changes.
+ *
+ * @author amajumdar
+ */
+@Slf4j
+@Component
+public class MViewServiceEventHandler {
+    private final Config config;
+    private final MViewService mViewService;
+    private final UserMetadataService userMetadataService;
+
+    /**
+     * Default constructor.
+     * @param config               server configurations
+     * @param mViewService         view service
+     * @param userMetadataService  user metadata service
+     */
+    @Autowired
+    public MViewServiceEventHandler(final Config config, final MViewService mViewService,
+        final UserMetadataService userMetadataService) {
+        this.config = config;
+        this.mViewService = mViewService;
+        this.userMetadataService = userMetadataService;
+    }
+
+    /**
+     * Subscriber.
+     *
+     * @param event event
+     */
+    @EventListener
+    public void metacatDeleteTablePostEventHandler(final MetacatDeleteTablePostEvent event) {
+        if (config.canCascadeViewsMetadataOnTableDelete() && !event.isMView()) {
+            final QualifiedName name = event.getTable().getName();
+            try {
+                // delete views associated with this table
+                final List<NameDateDto> viewNames = mViewService.list(name);
```

Comment:
any reason to even bother saving this to a local variable and not just doing the whole thing in one statement?